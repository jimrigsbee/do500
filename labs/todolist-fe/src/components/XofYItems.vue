<template>
    <div>
        <div class="xofyDone">
            <span> {{sumDoneTodoItems(todos)}} out of {{this.todos.length}} done. </span>
            <md-button class="md-raised" v-on:click="clearDoneTodos()">Clear Done</md-button>
            <md-button class="md-raised" v-on:click="clearTodos()">Clear all</md-button>
        </div>
    </div>
</template>

<script>
import { mapGetters } from "vuex";

export default {
  name: "XofYItems",
  computed: {
    ...mapGetters(["todos"])
  },
  created() {
    this.$store.dispatch("loadTodos");
  },
  methods: {
    clearDoneTodos() {
      this.$store.dispatch("clearTodos");
    },
    clearTodos() {
      // NOTE - true = all todos
      this.$store.dispatch("clearTodos", true);
    },
    sumDoneTodoItems(todos) {
      return todos.reduce(
        (result, tdItem) => (tdItem.completed ? result + 1 : result),
        0
      );
    }
  }
};
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped lang="scss">
.xofyDone {
  height: 52px;
  line-height: 52px;
  display: inline-block;
}
</style>
