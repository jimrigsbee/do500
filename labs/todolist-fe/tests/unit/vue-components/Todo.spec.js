/* eslint-disable */
import { shallow } from "@vue/test-utils";
import Todo from "@/views/Todo.vue";
import Header from "@/components/Header.vue";
import XofYItems from "@/components/XofYItems.vue";
import * as all from "../setup.js";

describe("Todo.vue", () => {
  it("should render the html layout like the snapshot (see /__snapshots__/Todo.spec.js.snap)", () => {
    const wrapper = shallow(Todo);
    expect(wrapper.element).toMatchSnapshot();
  });
  it("render with content such as header", () => {
    const wrapper = shallow(Todo);
    expect(wrapper.find(Header)).toBeTruthy();
  });
  it("render with content such as XofYItems", () => {
    const wrapper = shallow(Todo);
    expect(wrapper.find(XofYItems)).toBeTruthy();
  });
  // NEW TEST GOES HERE
});
