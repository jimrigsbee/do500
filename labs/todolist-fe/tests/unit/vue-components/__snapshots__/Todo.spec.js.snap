// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo.vue should render the html layout like the snapshot (see /__snapshots__/Todo.spec.js.snap) 1`] = `
<div>
  <!---->
</div>
`;
